api/src/main/java/org/apache/iceberg/metrics/ScanReport.java

```diff
@@ -83,12 +87,34 @@ public String toString() {
         .toString();
   }
 
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
```

Comment:
Style: missing whitespace between control flow and the following statements.

Reply:
equals() / hashCode() methods were generated